### PR TITLE
Update serialization cookbook to include Python APIs

### PR DESCRIPTION
**Pull request by GleasonK**
Move instructions on creating serialized artifacts to `compatibility.md` to have a centralized location to look for user compatibility documentation.

Also add Python APIs and augment with links to example code.

### CODE REVIEW

File: docs/compatibility.md [hunk 1]
```diff
@@ -46,6 +46,93 @@ guarantees, but please let us know if this is an important use case for you,
...
+See [`StablehloTranslateMain.cpp`](https://github.com/openxla/stablehlo/blob/main/stablehlo/tools/StablehloTranslateMain.cpp#:~:text=serializePortableArtifact)
+for example usage of these APIs.
+
+### Using Python APIs
+
+StableHLO also provides Python bindings to the C++ Serializaiton APIs:
+
+```python
+version = stablehlo.get_current_version()
+version = stablehlo.get_minimum_version()
```
Comment (burmako) [inline docs/compatibility.md:125]: This is a bit confusing. How about `# or alternatively: version = stablehlo.get_minimum_version()`, or something along these lines?

**Review by burmako — CHANGES_REQUESTED** [review]

File: docs/compatibility.md (hunk 1 above)
Comment (GleasonK) [reply docs/compatibility.md]: Used pyi format to describe APIs

File: docs/compatibility.md [hunk 2]
```diff
@@ -46,6 +46,93 @@ guarantees, but please let us know if this is an important use case for you,
...
+for example usage of these APIs.
+
+### Using Python APIs
+
+StableHLO also provides Python bindings to the C++ Serializaiton APIs:
+
+```python
+def get_current_version() -> str: ...
+def get_minimum_version() -> str: ...
+def serialize_portable_artifact(ir.Module: module, target_version:str) -> bytes: ...
```
Comment (burmako) [inline docs/compatibility.md:126]: `module: ir.Module`?

Comment (burmako) [inline docs/compatibility.md:126]: Also, do we want to call out string-based overloads? I think we should.

**Review by burmako — CHANGES_REQUESTED** [review]

File: docs/compatibility.md [hunk 3]
```diff
@@ -46,6 +46,95 @@ guarantees, but please let us know if this is an important use case for you,
...
+
+StableHLO also provides Python bindings to the C++ Serializaiton APIs:
+
+```python
+def get_current_version() -> str: ...
+def get_minimum_version() -> str: ...
+def serialize_portable_artifact(module: ir.Module, target_version: str) -> bytes: ...
+def serialize_portable_artifact(module: str, target_version: str) -> bytes: ...
+def deserialize_portable_artifact(artifact: bytes) -> ir.Module: ...
+def deserialize_portable_artifact(artifact: bytes) -> str: ...
```
Comment (burmako) [inline docs/compatibility.md:129]: These two signatures only differ in return types. How does this work?

**Review by burmako — CHANGES_REQUESTED** [review]

File: docs/compatibility.md (hunk 3 above)
Comment (GleasonK) [reply docs/compatibility.md:129]: Looks like I didn't hit save on this file before pushing changes. Missing an argument in the first one.

**Pull request merged**